### PR TITLE
Interaction tic tac toe

### PR DESCRIPTION
**Pull request by onerandomusername**
## Relevant Issues

closes GH-976

## Description

switched to using interactions for tic tac toe
Update 19/12/2021:
The interface was redone, and the user now has a different interface.
I've edited the screenshots to be up to date
*start of game*
image

*in progress*
image

*game over, win*
image

*game over, tie*
image

## Did you:

- [x] Join the [**Python Discord Community**](https://discord.gg/python)?
- [x] Read all the comments in this template?
- [x] Ensure there is an issue open, or link relevant discord discussions?
- [x] Read and agree to the [contributing guidelines](https://pythondiscord.com/pages/contributing/contributing-guidelines/)?

### CODE REVIEW

File: bot/exts/fun/tic_tac_toe.py [hunk 1]
```diff
@@ -77,50 +91,58 @@ def __init__(self, bot_user: discord.Member, symbol: str):
         self.symbol = symbol
 
     @staticmethod
-    async def get_move(board: dict[int, str], _: discord.Message) -> tuple[bool, int]:
+    async def get_move(board: dict[int, str], msg: discord.Message) -> tuple[bool, int, None]:
```
Comment (decorator-factory) [inline bot/exts/fun/tic_tac_toe.py:94]: Does the interface really has to be the same?
You're already checking if the player is an `AI` or a `Player`. So maybe just do something like:
```py
if isinstance(self.current, AI):
    pos = await self.current.get_move(self.board, board)
    interaction = None
else:
    move = await self.current.get_mode(self.board, board)
    if move is None:
        # handle timeout
    pos, inter = move
```

File: bot/exts/fun/tic_tac_toe.py [hunk 2]
```diff
@@ -192,41 +214,70 @@ def format_board(self) -> str:
     async def play(self) -> None:
```
Comment (decorator-factory) [inline bot/exts/fun/tic_tac_toe.py:214]: I don't have any concrete suggestions here, but this method is pretty complicated. I don't understand how the game logic works

**Review by decorator-factory — CHANGES_REQUESTED** [review]

**Comment by Xithrius** [comment]
@onerandomusername will you be completing this PR?

**Comment by Xithrius** [comment]
I'll be closing this PR due to staleness. If this PR is going to be completed, please ping me, along with re-opening.

**Pull request closed without merging**